lib/modules/platform/gitlab/index.ts

```diff
@@ -675,32 +677,60 @@ async function tryPrAutomerge(
         const { body } = await gitlabApi.getJson<{
           merge_status: string;
           detailed_merge_status?: string;
-          pipeline: string;
+          pipeline: {
+            status: string;
+          };
         }>(`projects/${config.repository}/merge_requests/${pr}`, {
           memCache: false,
         });
+        // detailed_merge_status is available with Gitlab >=15.6.0
+        const use_detailed_merge_status = Object.hasOwn(
+          body,
+          'detailed_merge_status',
+        );
```

Comment:
```suggestion
        const use_detailed_merge_status = !!body.detailed_merge_status;
```
why not simply doing this? 🤔 this way typescript probably can infer not null on condition

Reply:
I double-checked with the Typescript documentation and I agree it should work as you're suggesting. I guess I'm just too biased from other programming languages 😃 .